lib/ansible/modules/web_infrastructure/letsencrypt.py

```diff
@@ -168,24 +168,39 @@
 import traceback
 from datetime import datetime
 
+from functools import wraps
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils._text import to_native
+from ansible.module_utils._text import to_native, to_text, to_bytes
 from ansible.module_utils.urls import fetch_url
 
 
+def _upsert_lowercase_headers(func):
+    '''
+     Decorator to append the lowercase representation of the header names (dict keys) returned in the info dict
+
+    '''
+    @wraps(func)
+    def func_wrapper(name, *args, **kwargs):
+        response, info = func(name, *args, **kwargs)
+        info.update(dict((header.lower(), value) for (header, value) in info.items()))
+        return response, info
+    return func_wrapper
+
+fetch_url = _upsert_lowercase_headers(fetch_url)
```

Comment:
I would just make this a plain function that calls fetch_url and then lowercases the keys instead of making it a decorator.  I'm a little hesitant about overriding the name (I'd use a different name than fetch_url) but I can see you do that so you don't have to change the name everywhere.  Since I'm not the module author, I won't block it over that but I think it's better style to use a new name.

Reply:
The reason I've used a decorator was indeed to prevent having to rename all fetch_url occurrences. And since I'm not altering the values but merely ensuring the lowercase representation is included, I figured a decorator would be easier. 
Maybe this should be fixed in the fetch_url module utility ? Otherwise every module that uses this function AND wants to support py3 needs to create a wrapper or create several conditionals to capture the case variations ?